Bootloader/Bootloader/Mcal/Dio/Dio_Internal.h

```diff
@@ -23,6 +23,8 @@
 /**************************************************************************************************
 *                                        MACRO DEFINITIONS                                        *
 **************************************************************************************************/
+#define DIO_NUMBER_OF_PORTS                                                                      3
+#define DIO_NUMBER_OF_CHANNELS                                                                   7
```

Comment:
Number of Ports should be 4, Number of channels should be 8 as well and related conditions should be changed